src/components/generator/generatorFunction.js

```diff
@@ -1,131 +1,147 @@
 import arePositionsSimilar from '../../logic/combos/similarPositions';
 
-export function findCombo(vars, retries) {
+/**
+ * Heuristically generates a random combo which adheres to the specified conditions.
+ * @param {*} conditions Conditions that the combo must meet
+ * @param {number} retries Number of times to retry when generating a combo fails
+ */
+export function generateRandomCombo(conditions, retries) {
   const [
-    tricks,
-    positions,
-    numberOfTricks,
-    startFromCheckbox,
-    startFromPosition,
-    allowDuplicates,
-    allowConsecutiveTricks,
-    allowSimilarPositions,
-    allowTransitions,
-    finishToFeet,
-    avgDifficulty,
-    maxDifficulty
-  ] = vars;
-
-  /**
-   * Returns true if the current trick is a legal one in regards to the
-   * specified conditions. This does not check for duplicates as it only
-   * looks at one trick and its predecessor in the combo.
-   */
-  const areLocalComboContitionsLegal = (index, lastTrick, currentTrick) => {
-    const isConsecutiveTricksLegal = allowConsecutiveTricks || lastTrick.id !== currentTrick.id;
-
-    const isEndAndStartEqual = currentTrick.startPos === lastTrick.endPos;
-    const isEndAndStartSimilar = arePositionsSimilar(currentTrick.startPos, lastTrick.endPos);
-    const isTransitionLegal = (allowSimilarPositions && isEndAndStartSimilar)
-                              || isEndAndStartEqual
-                              || allowTransitions;
-
-    if (!isConsecutiveTricksLegal || !isTransitionLegal) {
-      return false;
-    }
+    tricks, positions, numberOfTricks,
+    startFromCheckbox, startFromPosition,
+    allowDuplicates, allowConsecutiveTricks, allowSimilarPositions, allowTransitions, finishToFeet,
+    avgDifficulty, maxDifficulty
+  ] = conditions
+
+  if (!retries) retries = 0
```

Comment:
Consider adding a new scope here to avoid potential bugs when changing this in the future

```js
if (!retries) {
    retries = 0;
}
``` 
Alternatively, this could be rewritten as follows:
```js
retries ??= 0
```

Reply:
I'd propose yet another option: Set the default value of `retries` to 0 in the function header.